modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/init.jsp

```diff
@@ -81,6 +82,10 @@ page import="java.util.Set" %>
 <%
 DDMFormAdminDisplayContext ddmFormAdminDisplayContext = (DDMFormAdminDisplayContext)request.getAttribute(WebKeys.PORTLET_DISPLAY_CONTEXT);
 
+HttpServletResponse httpServletResponse = PipingServletResponse.createPipingServletResponse(pageContext);
+
+ddmFormAdminDisplayContext.setHttpServletResponse(httpServletResponse);
+
```

Comment:
Hey @marcosapmf,

Is there any way of keep this logic on the back-end too? I mean, move this logic to `DDMFormAdminDisplayContext.java`. Have you checked if we can get the page context in the back-end?

Reply:
Hi @natocesarrego,

I was looking for a way to that but `pageContext` is specific to JSP. 

We could do that by creating a `setPageContext` method into `DDMFormAdminDisplayContext.java` but I think it's unnecessary once I just need the `pageContext` to create a `httpServletResponse` instance by `PipingServletResponse`.

Reply:
Hi @marcosapmf,

I see your point. But, what looks unusual to me is we have methods `get` and `set` manipulating the same attribute and behaving differently. I mean, why the `get` method is private and the `set` method is public, does is make sense? What about the encapsulation of the data?

Additionally, we're a not the first guys requesting the `pageContext` in the back-end, please check the method [WorkflowDefinitionDisplayContext.getCreationMenu](https://github.com/liferay/liferay-portal/blob/53dc2ed1278ebe0004de1868edab4f7c2f444d2c/modules/apps/portal-workflow/portal-workflow-web/src/main/java/com/liferay/portal/workflow/web/internal/display/context/WorkflowDefinitionDisplayContext.java#L136-L158) and its invocation from a JSP [here](https://github.com/liferay/liferay-portal/blob/53dc2ed1278ebe0004de1868edab4f7c2f444d2c/modules/apps/portal-workflow/portal-workflow-web/src/main/resources/META-INF/resources/definition/view.jsp#L40). In light of this, I believe that your suggestion of create a method to set the `pageContext` in the back-end isn't a bad idea. 🙂 Think about it and please send me a new pull.